bip-0360.mediawiki

```diff
@@ -11,696 +11,397 @@
-to write a tapscript which will duplicate stack elements until it reaches the maximum number of elements on stack, i.e. 1000 elements. 
-An increase from 520 bytes to 8,000 bytes would increase the memory footprint from 520 KB to 8 MB.
+Design of the P2TSH output type is guided by the following intentions:
 
-To prevent OP_DUP from creating an 8 MB stack by duplicating stack elements larger than 520 bytes we define OP_DUP to fail on stack 
-elements larger than 520 bytes. Note this change to OP_DUP is not consensus critical and does not require any sort of fork. This is
-because currently there is no way to get a stack element larger than 520 bytes onto the stack so triggering this rule is currently
-impossible and would only matter if the stack element size limit was raised.
+1. Minimize changes to the network. We should reuse existing Bitcoin code and preserve existing software behavior, workflows, user expectations and compatibility whenever possible.
 
-==== Public keys larger than 520 bytes ====
+P2TSH leverages the battle tested P2TR, tapleaf and tapscript code already in Bitcoin, reducing the implementation burden on wallets, exchanges, and libraries that can reuse code they already have. This approach reduces complexity and minimizes implementation risks.
 
-Turning our attention to public keys larger than 520 bytes. This is not needed for SLH-DSA as its public key is only 32 bytes. 
-This is a different problem than signatures as public keys are typically pushed onto
-the stack by the script (redeem script) to commit to public keys in the output. The OP_PUSHDATA opcode in tapscript fails if asked to push
-more than 520 bytes onto the stack. 
+2. Create the safest possible path for the addition of post-quantum signature integrations, in the event that they are used in the future.
 
-To solve this issue, for signature schemes with public keys greater than 520 bytes, we use the hash of the public key in the script.
-We then package the public key and signature together as the same stack element on the input stack. Since the hash of the public key is
-only 32 bytes, the script can push it on the stack as it does today. Consider the following example with a
-OP_CHECKMLSIG opcode for ML-DSA:
+Importantly, we are proposing a tapscript-native output type that is resistant to long-exposure attacks. While some existing output types are already resistant to long-exposure attacks (e.g. P2WSH), no such output type supports tapscript - a feature that may be required for practical implementation of post-quantum signature opcodes.
 
-<source>
-stack = [pubkey||signature]
-tapscript = [OP_PUSHDATA HASH256(expected_pubkey), OP_CHECKMLSIG]
-</source>
+P2WSH, for instance, is not tapscript-native and as such does not support the OP_SUCCESSx opcode update path that will be critical for the integration of post-quantum OP_CHECKSIG opcodes into Bitcoin.<ref><code>OP_SUCCESSx</code> is a mechanism to upgrade tapscript</ref>
```

Comment:
Some months ago it was discussed not to link this BIP to any others respecting introduction of new opcodes or anything else.